packages/teleport/src/Nodes/AddNode/useAddNode.ts

```diff
@@ -15,34 +15,50 @@ limitations under the License.
 */
 
 import { useState, useEffect } from 'react';
-import TeleportContext from 'teleport/teleportContext';
+import { formatDistanceStrict } from 'date-fns';
 import useAttempt from 'shared/hooks/useAttemptNext';
+import cfg from 'teleport/config';
+import TeleportContext from 'teleport/teleportContext';
+import { BashCommand, NodeToken } from 'teleport/services/nodes';
 
 export default function useAddNode(ctx: TeleportContext) {
   const { attempt, run } = useAttempt('processing');
-  const canCreateToken = ctx.storeUser.getTokenAccess().create;
   const version = ctx.storeUser.state.cluster.authVersion;
   const user = ctx.storeUser.state.username;
   const isAuthTypeLocal = !ctx.storeUser.isSso();
   const [automatic, setAutomatic] = useState(true);
   const [script, setScript] = useState('');
   const [expiry, setExpiry] = useState('');
+  const [token, setToken] = useState('');
 
   useEffect(() => {
     createJoinToken();
   }, []);
 
   function createJoinToken() {
     return run(() =>
-      ctx.nodeService.createNodeBashCommand().then(cmd => {
+      ctx.nodeService.fetchJoinToken().then(token => {
+        const cmd = createNodeBashCommand(token);
         setExpiry(cmd.expires);
         setScript(cmd.text);
+        setToken(token.id);
       })
     );
   }
 
+  function createNodeBashCommand(node: NodeToken): BashCommand {
+    const { expiry, id } = node;
+
+    const expires = formatDistanceStrict(new Date(), new Date(expiry));
+    const text = `sudo bash -c "$(curl -fsSL ${cfg.getNodeScriptUrl(id)})"`;
+
+    return {
+      text,
+      expires,
+    };
+  }
+
```

Comment:
if functions don't depend on prop states, then they should be defined outside of components, otherwise we add more work every time this component re-renders b/c the func gets redeclared (not that it is doing much atm and is a pre-optimization thinking, but it's a good habit to get into)